bqskit/passes/control/foreach.py

```diff
@@ -38,6 +38,11 @@ class ForEachBlockPass(BasePass):
     pass_down_key_prefix = 'ForEachBlockPass_pass_down_'
     """If a key exists in the pass data with this prefix, pass it to blocks."""
 
+    pass_down_block_specific_key_prefix = (
+        'ForEachBlockPass_specific_pass_down_'
+    )
+    """Key for injecting a block specific pass data."""
```

Comment:
The `pass_down_key_prefix` gets away easily because it doesn't have to maintain a parallel array with the internal block structure to the ForEach pass. Here, we do, so we should properly document:

1. What data type is expected here (`dict[int, Any]`)
2. What the `int` keys represent
3. How we construct the internal block array, emphasizing the ordering

Feel free to use a multi-line docstring here, and maybe even give a short informal example like you did in the PR comment.

Reply:
Changed the doc to include an example. I think simplifying what I put further might lead to confusion as to how to use the key, but it's pretty verbose right now. Let me know if you have suggestions about how it could be changed.